.github/workflows/rust-tests.yml

```diff
@@ -0,0 +1,45 @@
+---
+name: Rust Tests
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: >
+          Python version to currently test
+        type: string
+        required: true
+
+      runner:
+        description: >
+          Describes the system this workflow should run on.
+        type: string
+        required: true
+jobs:
+  unit-tests:
+    name: Tests-Rust-${{ inputs.python-version }}
+    runs-on: ${{ inputs.runner }}
+    timeout-minutes: 60
+    env:
+      QISKIT_SUPPRESS_PACKAGING_WARNINGS: Y
+      PIP_CACHE_DIR: ${{ github.workspace }}/.pip
+      QISKIT_TEST_CAPTURE_STREAMS: 1
+      HAVE_VISUAL_TESTS_RUN: false
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ inputs.python-version }}
+        id: python-outputs
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ inputs.python-version }}
+      - name: Install Qiskit directly
+        run: |
+          set -e
+          python -m pip install -c constraints.txt -e .
+      - name: Run Rust Tests
+        # Note that we use the virtual env with Qiskit installed to run the Rust
+        # tests since some of them still depend on Qiskit's Python API via PyO3.
```

Comment:
There is no virtual environment used in this job.
```suggestion
```

Reply:
Removed in fb635da